src/userdev/java/net/minecraftforge/gradle/userdev/MinecraftUserRepo.java

```diff
@@ -165,7 +165,13 @@ public void validate(Configuration cfg, Map<String, RunConfig> runs, File native
             vars.put("mcp_mappings", MAPPING);
 
             parent.getConfig().runs.forEach((name, dev) -> {
-                runs.computeIfAbsent(name, k -> new RunConfig()).merge(dev, false, vars);
+                List<RunConfig> runsToMerge = runs.values().stream().filter(run -> name.equals(run.getMerge())).collect(Collectors.toList());
+
+                if (runs.containsKey(name)) {
+                    runsToMerge.add(runs.get(name));
+                }
+
+                runsToMerge.forEach(run -> run.merge(dev, false, vars));
```

Comment:
This doesn't create if the build.gradle doesn't have it. Also it doesn't merge things that are not provided by the MDK.
What you probably want is to introduce a 'parent/child' system into to the run configs. And then walk the parents:

```
  client {
  }
  testClient {
    parent runs.client
  }
```

``` 
  run = runs.computeIfAbsent(name, new run);
   run.merge(dev);
   runs.forEach(e -> if (e.getParent() == null) propogate(run));

void propogate(run) {
  run.getChildren().forEach(child -> {
    child.merge(run);
    propogate(child)
  }
}
````